File: broker-shop/src/main/java/org/tyaa/demo/java/springboot/brokershop/models/ProductModel.java
package org.tyaa.demo.java.springboot.brokershop.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;

import java.math.BigDecimal;

@Data
@ToString(exclude = "image")
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ProductModel {
    private Long id;
    private String title;
    private String description;
    private BigDecimal price;
    private Integer quantity;
    private String image;
    private Long categoryId;
    private CategoryModel category;
}

File: broker-shop-frontend/src/models/ProductModel.ts
export default class ProductModel {
    public id: number
    public title: string
    public description: string
    public quantity: number
    public price: number
    public categoryId: number
    public image: string
    constructor (id: number, title: string, description: string, quantity: number, price: number, categoryId: number, image: string) {
        this.id = id
        this.title = title
        this.description = description
        this.quantity = quantity
        this.price = price
        this.categoryId = categoryId
        this.image = image
    }
}